### ISSUE TITLE
False positives with dynamic enums

### ISSUE THREAD

**Issue by onskulit** [webpro/knip#387]
Hello! 

In my project there are quite a few occurrences where we do not access the enum fields indirectly when they come from the backend. We describe the enum, use it in the response interface, mappers or whatever, and then, for example, just show this field. Accordingly, nowhere in the project is there a direct appeal to enum.

```
export enum CarBrand {
    BMW: 'BMW',
    Ford: 'Ford',
}

interface GetCarResponse {
    brand: CarBrand;
    owner: string;
}

const CarBrand: FC = ({ brand }: { brand: CarBrand }) => {
     return <h3>{brand}</h3>
}
```

It seems like in this situation we don't want to highlight the enum field as unused? Because the field comes from the backend and any of these fields can actually be used.

**Comment by onskulit** [comment]
> Does that help?

Thank you for your response! 

Unfortunately, it doesn't seem to be exactly what I'm looking for. The first point will not work because enum is used in another project file where I map server responses and describe types (my code example is completely synthetic). The second point won't work because I'm actually interested in this rule and there are a lot of really unused enum members

At this point, it’s probably all that’s really left to do is mark up individual cases manually. But conceptually it seems to me that knip should not mark any enum members unused if we can get any of those members dynamically from somewhere without accessing it directly (backend, localStorage e.t.c.). 

I will think about this and share if a solution appears in the development process. I thought it was worth highlighting this problem.

**Comment by webpro** [comment]
> knip should not mark any enum members unused if we can get any of those members **dynamically** from somewhere without accessing it directly (backend, localStorage e.t.c.).

Knip does **static** analysis. At this point I'm inclined to think they're not false positives :) But happy to discuss further or look into a reproduction of the issue if you think there's something Knip can or should do!

**Comment by webpro** [comment]
By the way, the other day I implemented supported for `ignoreBinaries` and `ignoreDependencies`. Planning to add similar support for `ignoreMembers` (mostly driven by e.g. Angular and React class components so `render()` and life-cycle methods can easily be ignored).

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.4.0. See [Release 5.4.0](https://github.com/webpro/knip/releases/tag/5.4.0) for release notes._
